### PR TITLE
Add User Guide & Design docs

### PR DESCRIPTION
**Pull request by hbinkim**

### CODE REVIEW

File: docs/USERGUIDE.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
+# User Guide
+
+## Design
+
+Hypersds-operator manages a custom resource (CR), `CephCluster` to manage an external ceph cluster which is located outside of Kubernetes cluster. Hypersds-operator watches `CephCluster` CR to create hypersds-provisioner which installs and updates the external ceph cluster.
+
+![hypersds-operator project overall diagram](assets/overall.png)
+
+## Workflow
```
Comment (hyoung-90) [inline docs/USERGUIDE.md:9]: The workflow part need to be updated.
It also need role and rolebinding for provisioner pod. The image has been updated.

File: docs/USERGUIDE.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+## Install Hypersds-Operator
+
+> Note: hypersds-operator image will be available on this [public registry](https://quay.io/organization/tmaxanc) when the official release is ready. Until then local image registry is needed to deploy and use hypersds-operator.
+
+``` shell
+# Download hypersds-operator project 
+$ git clone https://github.com/tmax-cloud/hypersds-operator.git
+$ cd hypersds-operator/
+
+# Deploy CRD
```
Comment (hyoung-90) [inline docs/USERGUIDE.md:30]: `make deploy` includes this task.
It is only needed when running on local.

File: docs/USERGUIDE.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+Hypersds-operator creates two configmaps and a secret to manage an external ceph cluster. Install configmap is based on the `CephCluster` CR spec. It contains ceph configuration data for bootstrapping external ceph cluster including the number of ceph cluster monitor daemon and etc. Access configmap is for accessing external ceph cluster when the bootstrap is completed. Hypersds-provisioner updates cluster access information and ceph keyring secret after the bootstrap.
+
+![hypersds-operator workflow](assets/workflow.png)
+
+## Before you begin
+
+- deploy K8s cluster through [minikube](https://kubernetes.io/docs/tasks/tools/install-minikube/), [kubeadm](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/install-kubeadm/), [kubespray](https://kubernetes.io/docs/setup/production-environment/tools/kubespray/), or [other methods](https://kubernetes.io/docs/setup/)
+
+> Note: hypersds CSI driver is needed to provision PVs and PVCs. Hypersds-operator will deploy hypersds CSI and this feature will be added soon.
```
Comment (hyoung-90) [inline docs/USERGUIDE.md:19]: The csi drive seems to be provided as guide.

**Comment by hbinkim** [comment]
Now it's ready for the final review.

**Comment by hyoung-90** [comment]
LGTM

**Pull request merged**